src/Moq/Sponsorships/Collections/DataCollection.cs

```diff
@@ -0,0 +1,151 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Moq.Sponsorships
+{
+
+    /// <summary>
+    /// A collection of user's personally identifiable information (PII)
+    /// which can be easily serialized and transmitted to other entities.
+    /// </summary>
+    public class DataCollection
+    {
+        /// <summary>
+        /// An arbitrary key-value pair of user personal information
+        /// </summary>
+        public abstract class DataEntry
+        {
+            /// <summary>
+            /// The key used to identify this data entry.
+            /// </summary>
+            public string Name { get; private set; }
+
+            /// <summary>
+            ///
+            /// </summary>
+            /// <param name="name"></param>
+            public DataEntry(string name)
+            {
+                Name = name;
+            }
+
+            /// <summary>
+            /// Sanitize key and append to value
+            /// Note that "value" is not itself sanitized, and can escape serialization.
+            /// </summary>
+            /// <param name="key"></param>
+            /// <param name="value"></param>
+            /// <returns></returns>
+            protected string Encode(string key, string value)
+                => $"{Sanitize(key)}: {value}";
+
+            /// <summary>
+            /// Encode a string in base64 to keep it from accidentally
+            /// making distinguishing different types of PII difficult.
+            /// </summary>
+            /// <param name="input">The string to encode</param>
+            /// <param name="originalEncoding">The encoding that the string is using; used to convert the input into a byte array.</param>
+            /// <returns></returns>
+            protected string Sanitize(string input, Encoding originalEncoding)
+                => Convert.ToBase64String(  originalEncoding.GetBytes(input), Base64FormattingOptions.None);
```

Comment:
I would suggest more secured way
```suggestion
            protected string Sanitize(string input, Encoding originalEncoding)
                => input.ToString();
```

Reply:
Also keeps the data open, in the spirit of OSS, LGTM